src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+        0,
+        &aNSM_talk_request,
+        &aNSM_talk_init,
+        &aNSM_talk_end_chk,
+        TRUE 
+    };
+
+    NPC_SHOP_MASTER_ACTOR *shop_master = (NPC_SHOP_MASTER_ACTOR *)actorx;
+    int action = 0;
+
+    CLIP(npc_clip)->ct_proc(actorx, game, &ct_data);
+
+    shop_master->npc_class.draw.main_animation.keyframe.morph_counter = 0.0f;
+    shop_master->sell_item = EMPTY_NO;
+    shop_master->npc_class.condition_info.hide_flg = FALSE;
+    shop_master->_9C0 = -1;
+    actorx->shape_info.draw_shadow = TRUE;
+
+    if (Common_Get(door_data).door_actor_name == RSV_NO) {
+        shop_master->npc_class.talk_info.melody_inst = 0;
+        action = 0x3d; 
+    } else {
+        action = 0;
+    }
+
+    aNSC_setupAction(shop_master, (GAME_PLAY *)game, action);
+}
+
+static void aNSM_actor_save(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->save_proc)(actorx, game);
+}
+
+static void aNSM_actor_dt(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->dt_proc)(actorx, game);
+}
+
+static void aNSM_actor_init(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->init_proc)(actorx, game);
+}
+
+static void aNSM_actor_draw(ACTOR* actorx, GAME* game) {
+    (*Common_Get(clip).npc_clip->draw_proc)(actorx, game);
+}
+
+static int aNSM_get_zone(xyz_t *wpos) {
+    if (wpos->x < (f32)120.0) {
+        if (wpos->z < (f32)180.0) {
+            return 1;
+        } else {
+            return 2;
+        }
+    }
+    if (wpos->z < (f32)160.0) {
+        return 0;
+    } else {
+        return 3;
+    }
+}
+
+static u8 aNSM_get_next_zone(u8 p1, u8 p2) {
+    static u8 next_zone[16] = {0, 1, 1, 1, 0, 1, 2, 2, 1, 1, 2, 3, 2, 2, 2, 3};
+    int t = p2 * 4;
+    return next_zone[t + p1];
+}
+
+static void aNSM_search_player(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *play) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    PLAYER_ACTOR *player = get_player_actor_withoutCheck(play); 
+
+    if (player != NULL) {
+        s16 angle = actorx->shape_info.rotation.y - shop_master->player_angle;
+        if (ABS(angle) > 0x4000) {
+            aNSC_setupAction(shop_master, play, 0x42);
+        } else {
+            chase_angle(&actorx->shape_info.rotation.y, shop_master->player_angle, 0x800);
+            actorx->world.angle.y = actorx->shape_info.rotation.y;
+        }
+    }
+}
+
+static void aNSM_search_player2(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *play) {
+    static f32 posX[4] = { 220.0, 100.0, 100.0, 220.0 };
+    static f32 posZ[4] = { 140.0, 140.0, 220.0, 220.0 };
+    PLAYER_ACTOR *player = get_player_actor_withoutCheck(play);
+    if (player != NULL) {
+            f32 dz = posX[shop_master->_9A3] - shop_master->npc_class.actor_class.world.position.x;
+            f32 dx = posZ[shop_master->_9A3] - shop_master->npc_class.actor_class.world.position.z;
+
+            s16 angle = atans_table(dx, dz);
+            chase_angle(&shop_master->npc_class.actor_class.shape_info.rotation.y, angle, 0x800);
```

Comment:
Same deal with the binary angle here. 0x800 is DEG2SHORT_ANGLE(11.25f). You can calculate the float angle by doing (binary angle / 32768.0f) * 180.0f.